### ISSUE TITLE
Adoption of JoJoJet/bevy-trait-query

### ISSUE THREAD

**Issue by Inspirateur** [bevyengine/bevy#15970]
Trait queries are missing in bevy and only covered by https://github.com/JoJoJet/bevy-trait-query/ 
Upstreaming it would be nicer for bevy users as it is a really common need and 3rd party lib can easily become outdated🥺 

(Got told by @alice-i-cecile to open this :p)

**Comment by giocri** [comment]
yeah this would be immensely useful for my game to i have several types of objects who have a common trait to write to a buffer and would have to do so in the right order which is an absolute pain to do without being able to do them all in the same system

**Comment by alice-i-cecile** [comment]
#20959 is another case where this would be tremendously useful at the engine level IMO.
